RickAndMorty-iOS/Views/Characters/CharactersViewController.swift

```diff
@@ -33,16 +47,64 @@ class CharactersViewController: BaseViewController {
         viewModel.refresh()
     }
 
-    override func loadView() {
-        view = charactersGridView
-        navigationController?.navigationBar.prefersLargeTitles = true
-        title = K.Titles.characters
-        navigationItem.rightBarButtonItem = charactersGridView.filterButton(self, action: #selector(filterButtonPressed))
-        navigationItem.leftBarButtonItem = charactersGridView.logoView()
+    override func viewWillAppear(_ animated: Bool) {
+        if viewOrientation == K.Orientation.landscape {
+            setupFilterButton(bottom: 5, height: 25)
+        }
+    }
 
-        charactersGridView.collectionView.delegate = self
-        charactersGridView.collectionView.refreshControl = UIRefreshControl()
-        charactersGridView.collectionView.refreshControl?.addTarget(self, action: #selector(onRefresh), for: .valueChanged)
+    override func viewWillDisappear(_ animated: Bool) {
+        removeFilterButton()
+    }
+
+    private func setupFilterButton(bottom: Int = 17, height: Int = 30) {
+        let rightButton = charactersGridView.filterButton(self, action: #selector(filterButtonPressed)).customView!
+        rightButton.tag = 1
+        navigationController?.navigationBar.addSubview(rightButton )
+        let targetView = self.navigationController?.navigationBar
+        rightButton.snp.makeConstraints({ make in
+            make.height.equalTo(height)
+            make.width.equalTo(80)
+            make.trailing.equalTo(targetView?.snp_trailingMargin ?? view.snp_trailingMargin).inset(5)
+            make.bottom.equalToSuperview().inset(bottom)
+        })
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        removeFilterButton()
+        if UIDevice.current.orientation.isLandscape {
+            setupFilterButton(bottom: 5, height: 25)
+            viewOrientation = K.Orientation.landscape
+        }
+
+        if UIDevice.current.orientation.isPortrait {
+            setupFilterButton(bottom: 17, height: 30)
+            viewOrientation = K.Orientation.potrait
+        }
+    }
```

Comment:
This function will still be called in the next screens such as Character Details View. It will make the filter button appear on those screens if users rotate their devices.